client/my-sites/domains/domain-management/edit/domain-types/mapped-domain-type.jsx

```diff
@@ -0,0 +1,360 @@
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+import { isSubdomain } from 'lib/domains';
+import { MAP_EXISTING_DOMAIN, MAP_SUBDOMAIN } from 'lib/url/support';
+import RenewButton from 'my-sites/domains/domain-management/edit/card/renew-button';
+
+class MappedDomainType extends React.Component {
+	getVerticalNavigation() {
+		return (
+			<VerticalNav>
+				{ this.emailNavItem() }
+				{ this.dnsRecordsNavItem() }
+				{ this.domainConnectMappingNavItem() }
+				{ this.transferMappedDomainNavItem() }
+			</VerticalNav>
+		);
+	}
+
+	emailNavItem() {
+		const path = emailManagement( this.props.selectedSite.slug, this.props.domain.name );
+
+		return <VerticalNavItem path={ path }>{ this.props.translate( 'Email' ) }</VerticalNavItem>;
+	}
+
+	dnsRecordsNavItem() {
+		const path = domainManagementDns( this.props.selectedSite.slug, this.props.domain.name );
+
+		return (
+			<VerticalNavItem path={ path }>{ this.props.translate( 'DNS Records' ) }</VerticalNavItem>
+		);
+	}
+
+	domainConnectMappingNavItem() {
+		const { supportsDomainConnect, hasWpcomNameservers, pointsToWpcom } = this.props.domain;
+		if ( ! supportsDomainConnect || hasWpcomNameservers || pointsToWpcom ) {
+			return;
+		}
+
+		const path = domainManagementDomainConnectMapping(
+			this.props.selectedSite.slug,
+			this.props.domain.name
+		);
+
+		return (
+			<VerticalNavItem path={ path }>
+				{ this.props.translate( 'Connect Your Domain' ) }
+			</VerticalNavItem>
+		);
+	}
+
+	transferMappedDomainNavItem() {
+		const { domain, selectedSite, translate } = this.props;
+
+		if ( domain.expired || domain.isSubdomain || ! domain.isEligibleForInboundTransfer ) {
+			return null;
+		}
+
+		const path = domainTransferIn( selectedSite.slug, domain.name, true );
+
+		return (
+			<VerticalNavItem path={ path }>
+				{ translate( 'Transfer Domain to WordPress.com' ) }
+			</VerticalNavItem>
+		);
+	}
+
+	resolveStatus() {
+		const { domain, translate, moment } = this.props;
+		const { expiry } = domain;
+
+		if ( isExpiringSoon( domain, 30 ) ) {
+			const expiresMessage = translate( 'Expires in %(days)s', {
+				args: { days: moment.utc( expiry ).fromNow( true ) },
+			} );
+
+			if ( isExpiringSoon( domain, 5 ) ) {
+				return {
+					statusText: expiresMessage,
+					statusClass: 'status-error',
+					icon: 'info',
+				};
+			}
+
+			return {
+				statusText: expiresMessage,
+				statusClass: 'status-warning',
+				icon: 'info',
+			};
+		}
+
+		if ( ! domain.pointsToWpcom ) {
+			return {
+				statusText: translate( 'Action required' ),
+				statusClass: 'status-error',
+				icon: 'info',
+			};
+		}
+
+		return {
+			statusText: translate( 'Active' ),
+			statusClass: 'status-success',
+			icon: 'check_circle',
+		};
+	}
+
+	renderExpiringSoon() {
+		const { domain, translate, moment } = this.props;
+		const { expiry } = domain;
+
+		if ( ! isExpiringSoon( domain, 30 ) ) {
+			return null;
+		}
+
+		if ( domain.bundledPlanSubscriptionId ) {
```

Comment:
The rendered components in both branches are basically the same - with some changes to properties and the main text. Maybe we could change the text based on the branch and then have one definition of what has to be rendered? I'm worried otherwise at some point these components will go out of sync.
Something like:
```
if ( domain.bundledPlanSubscriptionId ) {
  text = "asfasdf";
  renewButtonProperties = {
     primary: true,
     customLabel: "Fjlaksjdfa",
  };
} else {
  // ....
}

return (
   <div><p>{ text }</p>
       <RenewButton
         selectedSite={ this.props.selectedSite }
         { ...renewButtonProperties } />
   </div>
);
```